astropy/stats/circstats.py

```diff
@@ -0,0 +1,448 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+"""
+This module contains simple functions for dealing with circular statistics, for
+instance, mean, variance, standard deviation, correlation coefficient, and so
+on. This module also cover tests of uniformity, e.g., the Rayleigh and V tests.
+The Maximum Likelihood Estimator for the Von Mises distribution along with the
+Cramer-Rao Lower Bounds are also implemented. Almost all of the implementations
+are based on reference [1], which is also the basis for the R package
+'CircStats' [2].
+"""
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+import numpy as np
+from astropy.units import Quantity
+from astropy import units as u
+
+__all__ = ['circmean', 'circvar', 'circmoment', 'circcorrcoef', 'rayleightest',
+           'vtest', 'vonmisesmle']
+__doctest_requires__ = {'vtest': ['scipy.stats']}
+
+
+def _components(data, w=None, p=1, phi=0.0, axis=None):
+    # Utility function for computing the generalized rectangular components
+    # of the circular data.
+    if w is None:
+        w = np.ones(data.shape)
+    elif (w.shape != data.shape):
+        raise ValueError('shape of w %s is not equals shape of data %s.'
+                         % (w.shape, data.shape,))
+
+    C = np.sum(w * np.cos(p * (data - phi)), axis)/np.sum(w, axis)
+    S = np.sum(w * np.sin(p * (data - phi)), axis)/np.sum(w, axis)
+
+    return C, S
+
+
+def _angle(data, w=None, p=1, phi=0.0, axis=None):
+    # Utility function for computing the generalized sample mean angle
+    C, S = _components(data, w, p, phi, axis)
+
+    # theta will be an angle in the interval [-np.pi, np.pi)
+    # [-180, 180)*u.deg in case data is a Quantity
+    theta = np.arctan2(S, C)
+
+    if isinstance(data, Quantity):
```

Comment:
I would not worry about that: the user can change it to degrees if they want to; all we care for here is that it is an angle unit. (If you do think this is important then I would for any quantity do `theta = theta.to(data.unit)`, so that the user gets the same unit back; after all, it could be arcseconds!).